lib/three_scale/tenant_id_integrity_checker.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+module ThreeScale
+  class TenantIDIntegrityChecker
```

Comment:
I found this class hard to understand, the code looks pretty obfuscated for me. I think it'd be good to add a header comment explaining the big picture. Like:

```
# This class looks for inconsistencies in `tenant_id`
#
#   1. Goes through all models having a `tenant_id` attribute
#   2. For each model, goes through all its associations
#   3. If any association has a `tenant_id` attribute, it verifies
#      the `tenant_id` in the association is the same as in the model.
```

Reply:
There is a single public method `#check`. I refactored it further to make it easier to follow. I don't like vague comments when the code can be made easier to follow. Let me know if it is still think it needs improvement.